### PR TITLE
feat: add bounties pallet

### PR DESCRIPTION
**Pull request by Roznovjak**
## Description
This PR adds the bounties pallet.
Extrinsic calls are not filtered out.

## Related Issue
#155

### CODE REVIEW

**Comment by Roznovjak** [comment]
I have tested a basic scenario: propose a bounty, approve the bounty, propose a curator, the curator accepts the invitation, close the bounty as finished, claim the bounty.

**Comment by jak-pan** [comment]
Bump runtime cargo.toml version and spec_version pls

File: runtime/Cargo.toml [hunk 1]
```diff
@@ -5,7 +5,7 @@ homepage = 'https://github.com/galacticcouncil/hydradx-node'
 license = 'Unlicense'
 name = 'hydra-dx-runtime'
 repository = 'https://github.com/galacticcouncil/hydradx-node'
-version = '3.0.0'
+version = '5.0.0'
```
Comment (green-jay) [inline runtime/Cargo.toml:8]: 3.1.0 rather?

Comment (jak-pan) [reply runtime/Cargo.toml:8]: I guess by introducing this, old nodes will not be compatible anymore. Since you're changing runtime itself.

Comment (Roznovjak) [reply runtime/Cargo.toml:8]: We have also slightly modified the configuration of the treasury pallet.

Comment (green-jay) [reply runtime/Cargo.toml:8]: ok nevermind, it is a breaking change

**Pull request merged**